### PR TITLE
Fixes for HP-UX 10.20

### PR DESCRIPTION
**Pull request by aaeegg**
With these changes, OpenSSL 1.1.1 compiles on HP-UX 10.20 (using gcc) and passes 'make test'.

I omitted any workarounds for the vendor-supplied compiler, HP ANSI C. That compiler has enough bugs that I think it's better to just use gcc.

The build environment used for testing was gcc 4.2.2, binutils 2.16.1, GNU Make 3.80, Perl 5.22.1, and zlib 1.2.7, and the following configuration: `./Configure hpux-parisc1_1-gcc --with-rand-seed=egd --with-zlib-include=/usr/local/include --with-zlib-lib=/usr/local/lib zlib enable-egd no-shared no-threads no-asm CPPDEFINES=OPENSSL_NO_SECURE_MEMORY CC=/usr/local/pa11_32/bin/gcc CXX=/usr/local/pa11_32/bin/g++`

### CODE REVIEW

**Comment by t8m** [comment]
This is rather a feature. But even if you regard it as a bug fix I do not think we want this kind of changes on 1.1.1 which is soon going out of support. Please note that for 1.1.1 we are currently committed doing security fixes only.

File: Configurations/unix-Makefile.tmpl [hunk 1]
```diff
@@ -517,7 +517,7 @@ clean: libclean
 	$(RM) $(PROGRAMS) $(TESTPROGS) $(ENGINES) $(SCRIPTS)
 	$(RM) $(GENERATED_MANDATORY) $(GENERATED)
 	-$(RM) `find . -name '*{- $depext -}' \! -name '.*' \! -type d -print`
-	-$(RM) `find . -name '*{- $objext -}' \! -name '.*' \! -type d -print`
+	find . -name '*{- $objext -}' \! -name '.*' \! -type d -exec $(RM) '{}' \;
```
Comment (tom-cosgrove-arm) [inline Configurations/unix-Makefile.tmpl:520]: Why does this line need to be changed, but not the line above, which uses the similar construction? Note that this change will cause a lot more processes to be created, slowing down the execution of this target. In addition, the leading `-` has been lost, so that a non-zero exit status from this command will now stop execution of further commands.

File: crypto/bio/b_addr.c [hunk 2]
```diff
@@ -853,7 +853,7 @@ int BIO_lookup_ex(const char *host, const char *service, int lookup_type,
             }
 
             if (endp != service && *endp == '\0'
-                    && portnum > 0 && portnum < 65536) {
+                    && portnum >= 0 && portnum < 65536) {
```
Comment (bernd-edlinger) [inline crypto/bio/b_addr.c:856]: why this change?  isn't port 0 something special?

Comment (aaeegg) [reply crypto/bio/b_addr.c:856]: From the commit log: "Recognize 0 as a valid return value for strtol in BIO_lookup_ex, making platforms without getaddrinfo behave the same as those with getaddrinfo when BIO_lookup_ex() is called with service="0". This change is necessary for test_ssl_old to succeed on platforms without getaddrinfo."

I don't know what port 0 signifies in this context, but it must be valid, or else test_ssl_old wouldn't be calling BIO_lookup_ex("127.0.0.1","0",...).

File: Configurations/unix-Makefile.tmpl (hunk 1 above)
Comment (tom-cosgrove-arm) [reply Configurations/unix-Makefile.tmpl:520]: Yes, I would normally expect this to be done with `xargs`.

Since there are lines in the `Makefile` that use `xargs` already, that would seem the best approach (i.e. it wouldn't be adding a new dependency)

Comment (aaeegg) [reply Configurations/unix-Makefile.tmpl]: I amended the commit to use `xargs` for both lines, tested it and it works. `xargs` on HP-UX 10.20 defaults to `-s 2048`, so no explicit limit parameter to `xargs` is necessary.

Comment (tom-cosgrove-arm) [reply Configurations/unix-Makefile.tmpl]: Yes, I would expect `xargs` on a given system to work within any command-line length restrictions on that system

**Comment by aaeegg** [comment]
> This is rather a feature. But even if you regard it as a bug fix I do not think we want this kind of changes on 1.1.1 which is soon going out of support. Please note that for 1.1.1 we are currently committed doing security fixes only.

OpenSSL 1.0.1e compiles without modification on HP-UX 10.20, so this could be considered a fix for a regression.

**Comment by bernd-edlinger** [comment]
Sorry it is a way too invasive change at this time.
